### PR TITLE
[Chore] 2523: Use https://github.com/containerd/log for logging

### PR DESCRIPTION
**Pull request by TinaMor**
# PR Description

This PR replaces direct dependencies on github.com/sirupsen/logrus and github.com/containerd/containerd/log with https://github.com/containerd/log to streamline our logging dependencies.

We continue to use logrus.SetOutput() in [ocihook.go](https://github.com/TinaMor/nerdctl/blob/replace-logrus-package/pkg/ocihook/ocihook.go) because not all Logrus APIs are provided by the new library.

Github issues #2523

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -8,3 +8,7 @@ _output
 # vagrant
 /.vagrant
 Vagrantfile
+
+# IDEs and editors
+/.vs
+/.vscode
```
Comment (AkihiroSuda) [inline .gitignore:14]: This shouldn't be committed here

Comment (AkihiroSuda) [reply .gitignore:14]: ping @TinaMor 

Also, please squash commits

**Comment by aznashwan** [comment]
> We continue to use logrus.SetOutput() in [ocihook.go](https://github.com/TinaMor/nerdctl/blob/replace-logrus-package/pkg/ocihook/ocihook.go) because not all Logrus APIs are provided by the new library.

@TinaMor while I did not personally test this myself, it looks like in the current implementation of the [`containerd/log.L.Logger` struct field delegates to `logrust.Entry`](https://github.com/containerd/log/blob/main/context.go#L71), which has a [public `Logger` field](https://pkg.go.dev/github.com/sirupsen/logrus#Entry) which we could directly call [`SetOutput()`](https://pkg.go.dev/github.com/sirupsen/logrus#Logger.SetOutput) on as follows

```go
logger := log.GetLogger(log.L.Context)
logger.Logger.SetOutput(io.MultiWriter(stderr, logFile))
```

Considering that the current implementation of `containerd/log` simply delegates to the now-maintenance-mode `logrus` anyway, I don't believe it's that big of a deal either way in its current state.

**Review by profnandaa — APPROVED** [review]
LGTM. Also means, should the project want to switch from `logrus` to something else with the same API, it will just be changing in one place.

**Comment by yankay** [comment]
Thanks@TinaMor
/lgtm

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Comment by AkihiroSuda** [comment]
@aznashwan Do you want to open a follow-up PR?

**Comment by TinaMor** [comment]
@AkihiroSuda  I'm looking into @aznashwan's suggestion. Will do a followup PR

**Pull request merged**